homeworks/05-redux-basic/src/App.tsx

```diff
@@ -0,0 +1,62 @@
+import { ChangeEvent, FormEvent, useState, useEffect } from "react";
+import { useDispatch } from "react-redux";
+
+import { useBalance } from "./selectors";
+import { updateBalance, credit, subtractPercentage, debit } from "./reducer";
+import "./App.css";
+
+function App() {
+  const balance = useBalance();
+  const dispatch = useDispatch();
+  const [value, setValue] = useState("");
```

Comment:
The type of input field should be `number` not `string` and default value easily can be `0` because any operation with zero doesn't change the balance.